cmd/observer/node_builder/observer_builder.go

```diff
@@ -1232,8 +1276,51 @@ func (builder *ObserverServiceBuilder) BuildExecutionSyncComponents() *ObserverS
 			requesterDependable.Init(builder.ExecutionDataRequester)
 
 			return builder.ExecutionDataRequester, nil
-		})
+		}).
+		Component("execution data pruner", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			// by default, pruning is disabled
+			if builder.executionDataPrunerHeightRangeTarget == 0 {
+				return &module.NoopReadyDoneAware{}, nil
+			}
 
+			execDataDistributor.AddOnExecutionDataReceivedConsumer(func(data *execution_data.BlockExecutionDataEntity) {
```

Comment:
This works for the initial implementation so long as we keep a reasonably large number of blobs. It is still possible for the indexer to fall behind which would cause the node to crash. I opened https://github.com/onflow/flow-go/issues/6138 to improve on the synchronization between the services so pruning only happens after indexing is complete.

Reply:
@peterargue, got it, I will open new RP for #6138